### ISSUE TITLE
0.3.2-dev seems to break ALE's usage of signs

### ISSUE THREAD

**Issue by kevindashgit** [neovim/neovim#9388]
I was about to post an issue for ALE, but turns out its something to do with newer version (0.3.0-dev) of neovim.

I created a [demo](https://asciinema.org/a/B0E1dTyqwSwKTGYfi3D4VeS8B) of the issue.

os: osx 10.14.2
iterm: 3.2.6
terminal: 2.9.1
neovim: 0.3.2-dev .  (problemo)
neovim: 0.3.0-dev .  (no problemo!)

**Comment by mhinz** [comment]
I'm an author of a plugin that uses signs and everything works as expected. Actually, the way signs are set never changed between Vim and Nvim.

Maybe a displaying bug. Check `:sign place` if it contains signs that aren't actually shown.

That said, you should create an issue for ALE anyway, because the ALE developers know their code best and could quickly find out _when_ it breaks. That would help to create a minimal example.

Since then there is not much that we can do, since it's not even clear if it's a Nvim bug at all.

(SCNR: It's called "sign column", not "gutter".)

**Comment by bfredl** [comment]
Closing for now, it there comes more details that shows this in fact is a nvim core issue, we can reopen. Also there is quite a long (and imprecise) jump between 0.3.0-dev and 0.3.2-dev. Is the same ALE version used in both cases? Bisecting (nvim and/or ALE) could help identify the issue.

**Comment by kevindashgit** [comment]
>Is the same ALE version used in both cases?
Yes.

>Check :sign place if it contains signs that aren't actually shown.
```bash
:sign place                                                                                           
--- Signs ---                                                                                         
Signs for /Users/joeuser/temp/test.js:                                                         
    line=6  id=16961  name=ALEErrorSign
    line=1  id=16961  name=ALEErrorSign
    line=6  id=16961  name=ALEErrorSign
    line=8  id=16962  name=ALEErrorSign
    line=5  id=16961  name=ALEErrorSign
    line=7  id=16961  name=ALEErrorSign
```

Each of those signs *should* not be there.

**Comment by mhinz** [comment]
Sorry, but we don't have the time to look into plugin issues. Unless you have a minimal example or vimrc that reproduces the issue or any of the ALE developers confirms there's something wrong with Nvim, nothing will happen from our side.

With so many people using nvim 0.3.1 and plugins that set signs out there, it's just very unlikely that it's a bug in Nvim.

**Comment by kevindashgit** [comment]
Sorry maybe I wasn't clear - the issue is found in `neovim: 0.3.2-dev (HEAD-87b40f7)`

A minimum `.vimrc` (sorry thought I posted this)
```bash
" leader
let mapleader = ','
call functions#PlugLoad()
call plug#begin('~/.config/nvim/plugged')

Plug 'w0rp/ale'
" "ALE
" Plugin 'w0rp/ale'
"
let g:ale_fixers = {
\   'javascript': ['eslint'],
\}
let b:ale_linters = ['eslint']
nmap <silent><leader>aa :ALEFix<CR>

call plug#end()
```

If you still think its not worth chasing down - maybe we can sit on it. I'm not blocked or anything. Just trying to be helpful. 

I can open an issue in ALE and point them to this too.

**Comment by stsewd** [comment]
I saw the same issue when I updated Neovim around this week (I always use the nightly appimage). Not sure if this is an ALE issue (I didn't found any important recent changes there), I'll try to revert some plugins updates.

**Comment by mhinz** [comment]
Okay, I tried the above vimrc. I used this file:

```javascript
var num = 1;

if (typeof num === 'number') {
  console.log(`${num} is a number!`);
} else {
  console.log(`${num} is *not* a number!`);
  console.log('typeof num: ', typeof num);
}
```

..and this `.eslintrc` in the same directory..

```javascript
module.exports = {
    "extends": "standard"
};
```

.. everything works as expected for me with nvim v0.3.1 and master.

I get error signs for superfluous semicolons and when I remove those and save, the signs disappear as well.

**Comment by mhinz** [comment]
> I tested this with some previous versions of ALE, the problem persist. Then I tried with nvim-0.3.1 (appimage), and the problem disappeared

This is a bit unclear. Is the latest ALE working with the nvim 0.3.1 appimage?

**Comment by stsewd** [comment]
Sorry, yes, it is.

**Comment by mhinz** [comment]
Cool, thanks for testing that.

Well, then I stick to what I said earlier: Unless there is a reliable way to reproduce an error with the latest ALE and the latest nvim release (0.3.2 is to be released very soon) or master, there is nothing that we can do.

**Comment by stsewd** [comment]
I downloaded the latest appimage today (which I suppose includes the commits from master/yesterday master), the problem still persists. I also tested it from the linux binaries.

**Comment by stsewd** [comment]
`init.vim`

```vim
call plug#begin('~/.local/share/nvim/plugged')
Plug 'w0rp/ale'  " Async lint engine
call plug#end()

let g:ale_echo_msg_error_str = 'E'
let g:ale_echo_msg_info_str = 'I'
let g:ale_echo_msg_warning_str = 'W'
let g:ale_echo_msg_format = '[%linter%] %s [%severity%]'
let g:ale_virtualtext_cursor = 1
```

I'm using python with flake8 here, and actually I just opened a blank file, when writting `def test():` it complains of unexpected EOF, when adding `pass` the _markers_ are still visible.

```python
def test():
    pass
    # Should be 2 blank lines after the func
def test2():
    pass
```

```python
def test():
    pass
    # Should be 2 blank lines after the func


def test2():
    pass
```

**Comment by mhinz** [comment]
Okay, there's something wrong with our releases.

Not reproduced with:

- compiled from repo
- using homebrew

Reproduced with:

- all releases (macOS, Linux, AppImage)

**Comment by justinmk** [comment]
Does that include the 0.3.2 prerelease

**Comment by mhinz** [comment]
Yes.

**Comment by mhinz** [comment]
Okay, we have narrowed it down to a integer overflow so far.

ALE uses a sign offset of 100000 by default. The faulty versions will show other IDs instead in `:sign place`.

The max of int16_t is 32767. You can "fix" the faulty versions by using something like `let g:ale_sign_offset = 1000`.

Investigating further.

**Comment by bfredl** [comment]
Recently, the printing of sign IDs above `32767` was broken in `:sign place` output, and ALE used it to read existing signs back. Fixed by #9394.

**Comment by stsewd** [comment]
Thanks all for fixing this! :tada:
